### PR TITLE
Fix issue with ssh flags used with scp.

### PR DESCRIPTION
**Pull request by jraygauthier**
As the '-p' flag attempt to preserve file attributes
with scp, it was making the scp command fail in
contexts where preserving attributes is not possible
(this is without mentionning the '22' argument that
followed and which seems to be accepted silently).

Here's the error it fixes:

```
afrontend> uploading key ‘my_secret.key’...
afrontend> scp: /run/keys/my_secret.key: No such file or directory
afrontend> error: command ‘['scp', '-P', '22', '-i', '/run/user/1000/nixops-tmp4MXxgW/id_nixops-afrontend', '-p', '22', '-oControlPath=/run/user/1000/nixops-ssh-tmpakFvsQ/master-socket', '/run/user/1000/nixops-tmp4MXxgW/key-afrontend', u'root@10.233.1.2:/run/my_secret.key']’ failed on machine ‘afrontend’ (exit code 1)
error: activation of 1 of 1 machines failed (namely on ‘afrontend’)
```

See the `['-P', '22']` part followed not long after by wrong `['-p', '22']`.

Note that this is a prerequisite for #300

### CODE REVIEW

**Comment by domenkozar** [comment]
This doesn't call `get_ssh_flags` anymore, which is needed sometimes to pass the private ssh key

**Comment by jraygauthier** [comment]
Good point. It seems that default options were working fine for the localhost case. Bringing back
`get_ssh_flags` uncovered supplementary arguments propagation errors in the container backend.
This one should do the trick.

File: nixops/backends/container.py [hunk 1]
```diff
@@ -63,9 +63,9 @@ def get_ssh_flags(self, *args, **kwargs):
         flags = super(ContainerState, self).get_ssh_flags(*args, **kwargs)
         flags += ["-i", self.get_ssh_private_key_file()]
         if self.host == "localhost":
-            flags.extend(MachineState.get_ssh_flags(self))
+            flags.extend(MachineState.get_ssh_flags(self, *args, **kwargs))
```
Comment (domenkozar) [inline nixops/backends/container.py]: Why is this needed?

Comment (jraygauthier) [inline nixops/backends/container.py]: so that [MachineState.get_ssh_flags](https://github.com/jraygauthier/nixops/blob/aa967926bfe5668a869fe0e714037c8de2e90548/nixops/backends/__init__.py#L233) gets its expected `scp` flag as specified by the caller. Otherwise it will default to `False`.

I honestly find these var args parameters confusing, I'm not sure why there was a need for it in the first place but I tried to make do with it.

**Comment by domenkozar** [comment]
@jraygauthier could you rebase? I think we can merge this.

**Comment by jraygauthier** [comment]
@domenkozar This PR is no longer relevant now that #804 fixes the problem.
